lib/navigation/imu_displacement.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+#include "core/time.hpp"
+#include "core/timer.hpp"
+#include "core/types.hpp"
+
+namespace hyped::navigation {
+
+class ImuDisplacement {
+ public:
+  ImuDisplacement(const core::ITimeSource &time, const core::TimePoint initial_time);
+  /**
+   * @brief update the imu estimate for both displacement and velocity. Displacement is
+   * used for cross checking agreement with wheel encooders, velocity is used for us to
+   * estimate velocity in the case we have no direct velocity sensors.
+   *
+   * @param imu_acceleration kalman filtered single value for acceleration
+   * @param imu_timestamp time at which easurement was taken
+   */
+  void updateImuDisplacement(const core::Float imu_acceleration,
+                             const core::TimePoint imu_timestamp);
```

Comment:
I feel like we want something like the old `DataPoint` here; the sensor should timestamp things in the first instance and from then on the timestamp should be carried around without any possibility for error; this is quite important for the correctness of the whole subsystem.

Reply:
Will add a todolater. Will fox this once we have the data formats implemeted from sensors.

Reply:
Please get to that asap, shouldn't take long.